### ISSUE TITLE
helm-find-files: wrong-type-argument stringp nil

### ISSUE THREAD

**Issue by manuel-uberti** [emacs-helm/helm#1226]
I'm using latest Helm from MELPA on GNU Emacs `25.0.50.1 (x86_64-unknown-linux-gnu, GTK+ Version 3.18.2) of 2015-10-17`.

When I use `M-x helm-find-fles` and type the name of a new file to be visited, this is what I get hitting RET:

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  substitute-in-file-name(nil)
  helm-find-file-or-marked("/home/manuel/org/temp.org")
  helm-execute-selection-action-1()
  helm-execute-selection-action()
  helm-internal(helm-source-find-files "/home/manuel/org/" "Find Files or Url: " nil "^organizer\\.org" "*Helm Find Files*" nil "/home/manuel/org/#" nil)
  apply(helm-internal (helm-source-find-files "/home/manuel/org/" "Find Files or Url: " nil "^organizer\\.org" "*Helm Find Files*" nil "/home/manuel/org/#" nil))
  helm(helm-source-find-files "/home/manuel/org/" "Find Files or Url: " nil "^organizer\\.org" "*Helm Find Files*" nil "/home/manuel/org/#" nil)
  apply(helm (helm-source-find-files "/home/manuel/org/" "Find Files or Url: " nil "^organizer\\.org" "*Helm Find Files*" nil "/home/manuel/org/#" nil))
  helm(:sources helm-source-find-files :input "/home/manuel/org/" :case-fold-search smart :preselect "^organizer\\.org" :ff-transformer-show-only-basename t :default "/home/manuel/org/#" :prompt "Find Files or Url: " :buffer "*Helm Find Files*")
  helm-find-files-1("/home/manuel/org/" "^organizer\\.org")
  helm-find-files(nil)
  funcall-interactively(helm-find-files nil)
  call-interactively(helm-find-files nil nil)
  command-execute(helm-find-files)
```

**Comment by twillis** [comment]
just ran into this as well after upgrading.

**Comment by xuchunyang** [comment]
I can reproduce this as well. I believe it is caused by 2eec568.

**Comment by thierryvolpiatto** [comment]
Chunyang Xu <notifications@github.com> writes:

> I can reproduce this as well.

Could reproduce too.

> I believe it is caused by 2eec568.

Probably, will look into this tomorrow.

Thanks.

-- 
Thierry
https://emacs-helm.github.io/helm/

**Comment by manuel-uberti** [comment]
It is working now, thank you for the quick fix.

**Comment by Gastove** [comment]
I'm still encountering this, but only in certain circumstances, mostly with some (but not all) URLs. For instance:

https://google.com opens just fine
https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=cats throws `Wrong type argument: stringp, nil`

**Comment by thierryvolpiatto** [comment]
Ross Donaldson <notifications@github.com> writes:

> I'm still encountering this, but only in certain circumstances, mostly with some (but not all) URLs. For instance:
>
> https://google.com opens just fine
> https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=cats throws Wrong type argument: stringp, nil

It is working now, but not fully fixed though, I didn't find why
helm-marked-candidates returns the first url, but not the second, so
even if it is working now please keep it open until I resolve this fully.

Thanks.

-- 
Thierry
https://emacs-helm.github.io/helm/

**Comment by sk8ingdom** [comment]
I'm still running into this issue. When I mark multiple files in `dired`, execute `dired-do-rename`, and then exit with `C-g`,  I get the same argument.

**Comment by thierryvolpiatto** [comment]
@sk8ingdom your error is not relative to `helm-find-files`, but to `helm-read-file-name` and should have been fixed by 1e17cb1d.

Thanks.
